app/ports/management/commands/update-portinfo.py

```diff
@@ -37,13 +37,24 @@ def handle(self, *args, **options):
 
         # Using the received list of port names, find related JSON objects
         ports_to_be_updated_json = []
+        portnames_found_in_portindex = []
         for port in data['ports']:
             if port['name'].lower() in ports_to_be_updated:
                 ports_to_be_updated_json.append(port)
+                portnames_found_in_portindex.append(port['name'].lower())
+
+        # Find deleted ports
+        deleted_ports = []
+        for portname in ports_to_be_updated:
+            if portname.lower() not in portnames_found_in_portindex:
```

Comment:
This could easily be O(n^2) if you happen to update all ports (which did happen in the past, for example when the svn expansion was removed). Super inefficient, don't do it.

If you do this, at least use a dictionary rather than an array to check whether a port exists in a set.

That said, did you happen to notice that if your input data structure (at the moment `data['ports']`) was itself a dictionary, it would be sufficient to just check
```python
if port['name'] in data['ports']
```
as opposed to having to first transform and store the data in a different way?

I need to collect some thoughts to decide if I'm happy with the way this is done. I believe it's time to create unit tests for this part of functionality, as this might get tricky. A bunch of questions:
* The answer might be a provisional no, but could it be that somehow a list of ports that need as update accidentally contains a new port that was added one commit after the portindex was generated? Will you ever set the status of that port back to active / where?
* What happens when just a subport is removed?
* What happens when the full Portfile is deleted?
* What happens when a (sub)port is moved from one file to another?
* Does it work correctly when a port with the same name is added after a while (will it be set to active again)?

Could we cover these scenarios in unit tests?
I also need to check again the distinction between deleted and obsolete ports.

Reply:
I have now made use of a dictionary.


> * The answer might be a provisional no, but could it be that somehow a list of ports that need as update accidentally contains a new port that was added one commit after the portindex was generated? Will you ever set the status of that port back to active / where?

If due to some mess with git commits, a port gets marked as inactive, it will remain inactive until a change is made to the portfile and the app gets an opportunity to update it.

> * What happens when just a subport is removed?

The current functionality will detect when a support is deleted. I have also added a simple unitest for same, which can be made more complex.

> * What happens when the full Portfile is deleted?

Main port and all subports will be marked as deleted.

> * What happens when a (sub)port is moved from one file to another?

The previous port will be marked as deleted and new one will be created.

> * Does it work correctly when a port with the same name is added after a while (will it be set to active again)?

Yes, I have added a unitest also.

> Could we cover these scenarios in unit tests?

I have already covered only some, will cover all very soon.